### PR TITLE
Adding spotbugs.

### PR DESCRIPTION
**Pull request by conniey**
* Adds support for spotbugs.
* Specifies `Locale.ROOT` when the default is used.
* Changes variables to static when it is needed.
* Removing unused locals and switch-cases
* Specify character sets for Base64. It is a subset of US-ASCII.
* Creating bugs for spotbugs that cannot be resolved yet. #43 #44 #42

### CODE REVIEW

File: src/main/java/com/microsoft/azure/proton/transport/ws/impl/WebSocketImpl.java [hunk 1]
```diff
@@ -624,46 +620,16 @@ public void pop(int bytes) {
         public void close_head() {
             underlyingOutput.close_head();
         }
+    }
 
-        private final char[] hexDigits = {'0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'A', 'B', 'C', 'D', 'E', 'F'};
```
Comment (conniey) [inline src/main/java/com/microsoft/azure/proton/transport/ws/impl/WebSocketImpl.java:628]: methods are never used.

File: src/main/java/com/microsoft/azure/proton/transport/ws/impl/WebSocketImpl.java [hunk 2]
```diff
@@ -90,12 +90,7 @@ public WebSocketImpl(int customMaxFrameSize) {
 
     @Override
     public TransportWrapper wrap(final TransportInput input, final TransportOutput output) {
-        return new WebSocketSniffer(new WebSocketTransportWrapper(input, output), new PlainTransportWrapper(output, input)) {
-            protected boolean isDeterminationMade() {
-                _selectedTransportWrapper = _wrapper1;
-                return true;
-            }
-        };
+        return new WebSocketSnifferTransportWrapper(input, output);
```
Comment (conniey) [inline src/main/java/com/microsoft/azure/proton/transport/ws/impl/WebSocketImpl.java:93]: Anonymous class is below. Spotbugs was not a fan of anon classes.

File: src/main/java/com/microsoft/azure/proton/transport/ws/impl/Base64.java [hunk 3]
```diff
@@ -251,12 +253,12 @@ public static String encodeBase64StringLocal(byte[] dataValues) throws IllegalAr
 
         /* Codes_SRS_BASE64_21_010: [If the `dataValues` is empty, the encodeBase64StringLocal shall return a empty string.] */
         if (dataValues.length == 0) {
-            return new String();
+            return "";
```
Comment (mssfang) [inline src/main/java/com/microsoft/azure/proton/transport/ws/impl/Base64.java:256]: is this because "" has the same benefit as String.EMPTY?

**Review by mssfang — APPROVED** [review]
LGTM!

**Pull request merged**